docs/adr/0003-use-docker-volumes-for-improved-macos-performance.md

```diff
@@ -0,0 +1,102 @@
+# 3. Use Docker volumes for improved macOS performance
+
+Date: 2020-11-23
+
+## Context
+
+A common problem developers have experienced with GOV.UK Docker is that the
+[asset pipeline][] compilation steps can be slow, to the point that it was once
+common that users [were served 504 responses][504-fix] on initial attempts to
+start an application. This slowness in GOV.UK Docker appeared to be an
+amplification of [existing performance][govuk-frontend-performance] issues,
+however when GOV.UK apps began upgrading
+from the end-of-life Ruby Sass to Sassc we began to see problems that pointed
+to a performance issue within GOV.UK Docker.
+
+Sassc is known to [significantly improve the compilation time][sassc-speed] of
+Sass compared to Ruby Sass. When we updated [Content Publisher][] to Sassc
+found that the [application.scss][] file compiled in 20% of the time (~20s to
+4s) when running natively on macOS. However when this was run on GOV.UK Docker
+the performance was significantly less impressive, compiling in 62% of the
+time (~31s to ~19s).
+
+We discovered that the source of this performance issue was the
+[performance overhead of reading][docker-mac-read] from the
+[shared mount][govuk-mount] of application files on macOS. We considered a number
+of options of how this could be improved:
+
+1. Use [Docker volumes][volume] to store non-application code (such as caches or
+   dependencies) to benefit from faster I/O access.
+2. [Configure GOV.UK Docker to use NFS][nfs-docker] for improved I/O access to
```

Comment:
```suggestion
2. [Configure GOV.UK Docker to use NFS][nfs-docker] (an alternative driver for shared mounts) for improved I/O access to
```

Minor: we could give a bit more indication what this alternative means in practice.